File: language-testing/rot13/rot13.clj
(defn rot13 [x]
  "Takes a string and does rot13 'encryption' on it"
  (defn rot_char [c]
    (let [low (compare c \a)
          high (compare c \A)]
      (cond
        (<=  0 low  12) (char (+ (int c) 13))
        (<= 13 low  25) (char (- (int c) 13))
        (<=  0 high 12) (char (+ (int c) 13))
        (<= 13 high 25) (char (- (int c) 13))
        :else c)))
  (apply str (map rot_char x)))

(if *command-line-args*
  (println (clojure.string/join " " (map rot13 *command-line-args*)))
  (doseq [line (line-seq (java.io.BufferedReader. *in*))]
    (println (rot13 line))))

File: language-testing/rot13/Rot13.java
import java.io.*;

public class Rot13 {
  public static void main(String[] args) throws IOException {

    /* join and rot args */
    if (args.length > 0) {
      StringBuilder sb = new StringBuilder();
      sb.append(args[0]);
      for (int i = 1; i < args.length; ++i) {
        sb.append(" ");
        sb.append(args[i]);
      }
      rot13(sb.toString());

    /* rot for stdin */
    } else {
      BufferedReader in = new BufferedReader(new InputStreamReader(System.in));
      String line;
      while ((line = in.readLine()) != null) {
        rot13(line);
      }
    }
  }

  public static void rot13(String string) {
    char[] cstr = string.toCharArray();
    for (int i = 0; i < cstr.length; ++i) {
      if      ('a' <= cstr[i] && cstr[i] <= 'm') { cstr[i] += 13; }
      else if ('n' <= cstr[i] && cstr[i] <= 'z') { cstr[i] -= 13; }
      else if ('A' <= cstr[i] && cstr[i] <= 'M') { cstr[i] += 13; }
      else if ('N' <= cstr[i] && cstr[i] <= 'Z') { cstr[i] -= 13; }
    }
    System.out.println(new String(cstr));
  }
}
